crates/bevy_winit/src/converters.rs

```diff
@@ -64,180 +65,226 @@ pub fn convert_touch_input(
     }
 }
 
-pub fn convert_virtual_key_code(virtual_key_code: winit::event::VirtualKeyCode) -> KeyCode {
+pub fn convert_physical_native_key_code(
+    native_key_code: winit::keyboard::NativeKeyCode,
```

Comment:
I'm completely wrong to use `NativeKeyCode`, as "[`VirtualKeyCode`](https://docs.rs/winit/0.28.7/winit/event/enum.VirtualKeyCode.html) was replaced by [`Key`](https://docs.rs/winit/latest/winit/keyboard/enum.Key.html)" from [winit changelog](https://github.com/rust-windowing/winit/blob/master/CHANGELOG.md)

I encountered 1 big problem with using `Key`, which made impossible to know when a key was released (see for details: https://github.com/Vrixyz/bevy/pull/3)

Reply:
Do we use this internally at all? If not, I think we should cut this API and revisit later with a clearer goal in mind.

Reply:
The impact on users is very visible: mapping over NativeKeycode rather than VirtualKeyCode (now Key) means the user loses the keyboard layout mapping:
- the bug introduced is a user on azerty layout expecting to input "a" when typing on "a" will actually input "q"

Reply:
It' a bit complicated to follow because of the naming:
- bevy 0.12 uses KeyCode to refer to a key **with** layout, where winit 0.28 uses VirtualKeyCode to refer to that.
- bevy 0.12 uses ScanCode(u32) to refer to key **without** layout, so does bevy.

winit introduced a "KeyCode", an enum replacing ScanCode. Not to be confused with bevy 0.12 KeyCode.

Reply:
Thanks!

For exactness: scan codes are already removed by this PR.

I'm happy with that direction, I'll still insist on one last thing:
You're saying you doubt users would need it, but let's study our examples: 
- [example `fog`](https://github.com/bevyengine/bevy/blob/main/examples/3d/fog.rs#L226) displays to the user: "A / S - Move Start Distance" -> where for azerty keyboards, it should read "Q / S - Move Start Distance", because a user typing on A won't see anything.
- I think we agree the bug is actually in 0.12, where we should not display the visible key to the user, it just "happened" to be consistent with our underlying implementation.
- Still now a new user with azerty is likely to not understand how to use the examples, mislead by this incorrect hard-coded key instruction 🙁.

Reply:
> Still now a new user with AZERTY is likely to not understand how to use the examples, mislead by this incorrect hard-coded key instruction 🙁.

You're right. I think that's OK though. We can address any non-standard hardware cases later.

Long-term the engine should move towards adopting a "action"-centric model like [Steam Input][1] and [LWIM][2] (or adopting LWIM if that's possible).

In the meantime, I did comment below with a new suggestion that `KeyboardInput` can hold onto the entire `KeyEvent`, and that way "logical key" information is available if users want to go read it.

[1]: https://partner.steamgames.com/doc/features/steam_controller/concepts
[2]: https://github.com/Leafwing-Studios/leafwing-input-manager